pages/story/[id].tsx

```diff
@@ -1,8 +1,34 @@
 import { useRouter } from 'next/router'
+import { Box } from '@chakra-ui/react'
+import { get } from '../../lib/api/stories'
+import StoryDetail from '../../components/stories/StoryDetail'
+import Footer, { FooterSpace, Button } from '../../components/common/Footer'
 
 export default function StoryPage() {
   const router = useRouter()
-  const { id } = router.query
+  const id = router.query.id as string
 
-  return <p>Story {id}</p>
+  // If we came from the feed, go back on cancel. If not, navigate forward to the feed.
+  function handleCancel() {
+    router.query.back === 'true' ? router.back() : router.push('/feed')
+  }
+
+  return (
+    <Box>
+      <StoryDetail id={id} onCancel={handleCancel} />
+      <FooterSpace />
+      <Footer>
+        <Button>Share This Story</Button>
+      </Footer>
+    </Box>
+  )
+}
+
+export async function getServerSideProps({ params }) {
+  try {
+    const prefected = await get(params.id)
```

Comment:
```suggestion
    const prefetched = await get(params.id)
```

Reply:
Good spots, thanks!

Reply:
Oh, it seems this goes away with Mike's changes. There's always a prefetched story, so it's just called story now.